src/settings.ts

```diff
@@ -375,6 +377,19 @@ export var settings = {
    * - `"icon"` - Users can only use the choice item icon as a drag handle.
    */
   rankingDragHandleArea: "entireItem",
+
+  /**
+   * Specifies environment in which SurveyJS will exist
+   */
+
+  environment: {
```

Comment:
> > > > Hi, @dk981234 slightly_smiling_face Any updates?
> > > 
> > > 
> > > Hello,
> > > Sorry for the delayed response. I've pulled your changes and found out that applyTheme and popups don't work anymore. I think problem with settings' initial value. document.body is not available at script load time. I have requested the relevant changes which may help.
> > > Thanks, Dmitry SurveyJS Team
> > 
> > 
> > Hi Dmitry, thanks for the replay slightly_smiling_face I've applied the changes you suggested, and I was quite surprised to find out that `document.body` isn't accessible from settings, but it's quite logical looking at `index.html` from `examples/react` and import in the head.
> > ```
> >     <link rel="stylesheet" href="../../build/survey-react/defaultV2.css" />
> >     <script src="../../build/survey-react/survey.react.js"></script>
> >   </head>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Nevertheless, I've applied your suggestion and extended the setting with getters and setters, since they were necessary for the other parts of the code. IMO the `settings.environment` became a little bit messy, but I'm ok with that change as long as you are. Maybe a good approach would be to switch to something like
> > ```
> > settings.environment = () => ({ ... })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > should be much cleaner than the getter setter approach.
> 
> We can remove private properties from interface. For users it will be clean as it was before requested changes. What do you think about it?

Sure 🙂 But wouldn't this approach mess up the types for the end user?
```ts
  environment: {
    root: document,
    get rootElement(): HTMLElement | ShadowRoot {
      return document.body;
    },

    get popupMountContainer(): HTMLElement | string {
      return document.body;
    },
    svgMountContainer: document.head,
    stylesSheetsMountContainer: document.head,
  },
```